lib/components/util/styledIcon.tsx

```diff
@@ -0,0 +1,62 @@
+import { StyledIconBase } from '@styled-icons/styled-icon'
+import styled, { keyframes } from 'styled-components'
+
+interface Props {
+  flipHorizontal?: boolean
+  noMargin?: boolean
+  rotate90?: boolean
+  size?: string
+  spaceRight?: boolean
+  spin?: boolean
+  static?: boolean
+}
+
+const getFontSize = (size?: string) => {
+  switch (size) {
+    case '1.5x':
+      return '1.5em'
+    case '2x':
+      return '2em'
+    case '3x':
+      return '3em'
+    case '4x':
+      return '4em'
+    case '5x':
+      return '5em'
+    default:
+      return 'inherit'
+  }
+}
+
+const rotateAnimation = keyframes`
+  0% {
+    transform: rotate(0deg);
+  }
+  100% {
+    transform: rotate(360deg);
+  }
+`
+
+const StyledIconWrapper = styled.span<Props>`
+  top: 0.125em;
+  position: ${(props) => (props.static ? 'static' : 'relative')};
+  display: inline-flex;
+  align-self: center;
```

Comment:
Does this work with `inline-flex`? Do we always know what the parent display type will be?

Reply:
I mean, it worked to replace everywhere we have icons. I used inline-flex because flex helps center the icon, and inline helps it integrate into text the same way the font based icons did.